content/en/llm_observability/setup/sdk/python.md

```diff
@@ -0,0 +1,683 @@
+    span_context = LLMObs.export_span(span=None)
+    return completion
+{{< /code-block >}}
+
+
+### Submit evaluations
+
+`LLMObs.submit_evaluation()` can be used to submit your custom evaluation associated with a given span.
+
+#### Arguments
+
+The `LLMObs.submit_evaluation()` method accepts the following arguments:
+
+`span_context`
+: required - _dictionary_
+<br />The span context to associate the evaluation with. This should be the output of `LLMObs.export_span()`.
+
+`label`
+: required - _string_
+<br />The name of the evaluation.
+
+`metric_type`
+: required - _string_
+<br />The type of the evaluation. Must be one of "categorical" or "score".
+
+`value`
+: required - _string or numeric type_
+<br />The value of the evaluation. Must be a string (for categorical `metric_type`) or integer/float (for score `metric_type`).
+
+`tags`
+: optional - _dictionary_
+<br />A dictionary of string key-value pairs that users can add as tags regarding the evaluation. For more information about tags, see [Getting Started with Tags][9].
+
+### Example
+
+{{< code-block lang="python" >}}
+from ddtrace.llmobs import LLMObs
+from ddtrace.llmobs.decorators import llm
+
+@llm(model_name="claude", name="invoke_llm", model_provider="anthropic")
+def llm_call():
+    completion = ... # user application logic to invoke LLM
+    span_context = LLMObs.export_span(span=None)
+    LLMObs.submit_evaluation(
+        span_context,
+        label="harmfulness",
+        metric_type="score",
+        value=10,
+        tags={"evaluation_provider": "ragas"},
+    )
+    return completion
+{{< /code-block >}}
+
+## Advanced tracing
+
+### Tracing spans using inline methods
+
+For each span kind, the `ddtrace.llmobs.LLMObs` class provides a corresponding inline method to automatically trace the operation a given code block entails. These methods have the same argument signature as their function decorator counterparts, with the addition that `name` defaults to the span kind (`llm`, `workflow`, and so on) if not provided. These methods can be used as context managers to automatically finish the span once the enclosed code block is completed.
+
+#### Example
+
+{{< code-block lang="python" >}}
+from ddtrace.llmobs import LLMObs
+
+def process_message():
+    with LLMObs.workflow(name="process_message", session_id="<SESSION_ID>", ml_app="<ML_APP>") as workflow_span:
+        ... # user application logic
+    return
+{{< /code-block >}}
+
+### Persisting a span across contexts
+
+To manually start and stop a span across different contexts or scopes:
+
+1. Start a span manually using the same methods (for example, the `LLMObs.workflow` method for a workflow span), but as a plain function call rather than as a context manager.
+2. Pass the span object as an argument to other functions.
+3. Stop the span manually with the `span.finish()` method. **Note**: the span must be manually finished, otherwise it will not be submitted.
```

Comment:
```suggestion
3. Stop the span manually with the `span.finish()` method. **Note**: the span must be manually finished, otherwise it is not submitted.
```